overbooking/src/main/java/org/tudelft/blockchain/booking/Overbooking.java

```diff
@@ -55,26 +78,55 @@ public Response invoke(ChaincodeStub stub) {
 
     private Response book(ChaincodeStub stub, List<String> params) {
         try {
-            DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("dd-MM-YYYY");
-            LocalDateTime bookingStart = LocalDateTime.from(dateFormat.parse(params.get(0))),
-                bookingEnd = LocalDateTime.from(dateFormat.parse(params.get(1))),
-                availableFrom = LocalDateTime.from(dateFormat.parse(stub.getStringState("availableFrom"))),
-                availableTill = LocalDateTime.from(dateFormat.parse(stub.getStringState("availableTill")));
-
-            if (bookingStart.isAfter(availableFrom)
-                    && bookingEnd.isBefore(availableTill)) {
-                stub.putStringState("unavailableFrom", params.get(0));
-                stub.putStringState("unavailableTill", params.get(1));
-            } else
-                return newErrorResponse("Invalid booking dates !");
+            LocalDateTime bookingStart = convertFromString(params.get(0)),
+                          bookingEnd = convertFromString(params.get(1));
+
+            if(isBookable(stub, bookingStart, bookingEnd)){
+                doBooking(stub, bookingStart, bookingEnd);
+                return newSuccessResponse();
+            }
+            else {
+                //TODO
+                //Implement returning the unavailable dates
+                return newErrorResponse("Invalid booking dates!");
+            }
 
-            return newSuccessResponse();
         } catch (Throwable e) {
             return newErrorResponse(e);
         }
     }
 
+<<<<<<< HEAD:overbooking/src/main/java/Overbooking.java
+    private Boolean isBookable(ChaincodeStub stub, LocalDateTime bookingStart, LocalDateTime bookingEnd) {
+
+        while(bookingStart!=bookingEnd){
+
+            //Retrieve information about overall availability
+            if(stub.getState(this.convertToString(bookingStart))==BookingStatus.BOOKED){
+                return FALSE;
+            }
+            bookingStart.plusDays(1);
+        }
+
+        return TRUE;
+    }
+
+    private void doBooking(ChaincodeStub stub, LocalDateTime bookingStart, LocalDateTime bookingEnd){
+
+        while(bookingStart!=bookingEnd){
```

Comment:
```suggestion
         while(bookingStart != bookingEnd){
```
Please use the auto formatter of intellij - Ctrl + Alt + L

Reply:
Thanks for the heads up!